Games/types/Mafia/roles/cards/KillVisitorsWhileDead.js

```diff
@@ -0,0 +1,52 @@
+const Card = require("../../Card");
+const { PRIORITY_KILL_VISITORS_ENQUEUE_DEAD, PRIORITY_KILL_VISITORS } = require("../../const/Priority");
+
+module.exports = class KillVisitorsWhileDead extends Card {
+
+    constructor(role) {
+        super(role);
+
+        this.actions = [
+            {
+                priority: PRIORITY_KILL_VISITORS_ENQUEUE_DEAD,
+                run: function () {
+                    if (this.actor.alive)
+                        return;
+
+                    for (let action of this.game.actions[0])
+                        if (
+                            action.target == this.actor &&
+                            action.priority > this.priority &&
+                            !action.hasLabel("hidden")
+                        ) {
+                            if (!this.actor.role.data.visitors)
+                                this.actor.role.data.visitors = [];
+
+                            this.actor.role.data.visitors.push(action.actor);
+                        }
+                }
+            },
+            {
+                priority: PRIORITY_KILL_VISITORS,
+                labels: ["kill", "hidden"],
+                run: function () {

```

Comment:
As mentioned above, it would be nice to have "action whileDead" in future. You can make an enhancement ticket for that.

Reply:
done as #349